### PR TITLE
feat(protocol):  optimize bridge with removal of depositValue

### LINKED ISSUE #14384: A few Bridge improvement ideas
### Describe the feature request

- rename `Message.owner` to `Message.user` for clarity
- rename `Message.sender` to `Message.from`, also rename `Context.sender` to `Context.from` for clarity. now `from` represents the app on the source change to interact with Bridge for sending the messages
- remove `Message.depositValue` because depositing Ether to an address on the destchain can be achieved by setting `callValue` to a non-zero. So `depositValue` is not redundant.  Sending Ether can be done 1) setting the `to` to the recipient address and `data` to `""`, and `callValue` to the amount of ether, 2) setting the `to` to an app on the destination change, and `callValue` to the amount of ether, and inside the app, a function will further forward the ether to the recipient address encoded in `data`. Removing `Message.depositValue` will simply the `processMessage` code.
- In addition to the above idea, returning Ether to users for failed messages shall also be handled by the source app's `recallMessage` function, not the Bridge. This will further simply the core bridge code.
- rename `callValue` to `value`
- maybe rename `processingFee` to `fee`, and `refundAddress` to `refundTo`
- remove all duplicate code in tests (this is a very bad way of writing tests)

Additional context here.

### PR DESCRIPTION
**Pull request by dantaik**

### CODE REVIEW

File: packages/protocol/contracts/bridge/Bridge.sol [hunk 1]
```diff
@@ -231,14 +229,10 @@ contract Bridge is EssentialContract, IBridge, BridgeErrors {
      * Check if the Ether associated with the given message hash has been
      * released.
      * @param msgHash The hash of the message.
-     * @return Returns true if the Ether has been released, false otherwise.
```
Comment (dantaik) [inline packages/protocol/contracts/bridge/Bridge.sol:234]: recall message may release Ether, but please find and remove all the "release Ether" doc-comments, instead, let the IRecallableMessageSender to decide whether the Ether should be returned to message.owner or not.

File: packages/protocol/contracts/bridge/IBridge.sol [hunk 2]
```diff
@@ -8,13 +8,13 @@ pragma solidity ^0.8.20;
...
 /**
- * Recalling a message interface.
+ * An interface that all recallable message sender shall implement.
  */
 
 interface IRecallableMessageSender {
     function onMessageRecalled(IBridge.Message calldata message)
         external
-        returns (bytes4);
+        payable;
```
Comment (dantaik) [inline packages/protocol/contracts/bridge/IBridge.sol:17]: message.callValue will be send to this contract as well. This is the couterparty of invoke message on the destination chain.

File: packages/protocol/test/mBridge.t.sol [hunk 3]
```diff
@@ -25,6 +25,9 @@ import { LibBridgeStatus } from "../contracts/bridge/libs/LibBridgeStatus.sol";
  * @dev The code hash for the same address on L1 and L2 may be different.
  * @custom:security-contact hello@taiko.xyz
  */
+
+// TODO(dani): why we have this contract? Shall we simply sub-class the real
```
Comment (dantaik) [inline packages/protocol/test/mBridge.t.sol:29]: Please remove duplicate code

File: packages/protocol/test/mLibBridgeRecall.t.sol [hunk 4]
```diff
@@ -15,24 +15,18 @@ import { LibBridgeData } from "../contracts/bridge/libs/LibBridgeData.sol";
...
 
-interface VaultContract {
-    function releaseToken(IBridge.Message calldata message) external;
-}
 /**
  * This library provides functions for releasing Ether related to message
  * execution on the Bridge.
  */
 
+// TODO(dani): why we have this library? I really don't like duplicate code,
```
Comment (dantaik) [inline packages/protocol/test/mLibBridgeRecall.t.sol:23]: please remove duplicate code

**Comment by dantaik** [comment]
@adaki2004 please keep working on this PR

**Comment by dantaik** [comment]
@adaki2004 feel free to continue working on this PR and merge

**Review by adaki2004 — APPROVED** [review]

**Pull request merged**